src/common/Formatter.cc

```diff
@@ -296,6 +296,17 @@ void JSONFormatter::finish_pending_string()
   }
 }
 
+void JSONFormatter::add_value(std::string_view name, double val) {
+  CachedStackStringStream css;
+  if (std::isinf(val) || std::isnan(val)) {
+    *css << "null";
```

Comment:
I would prefer we use a string like "[+-]inf" and "NaN". Then it can be decoded if desired to the original representation.

Reply:
Are you looking for quoted string "[+-]inf"?

Reply:
the JSON RFC doesn't know things like `-inf` or similar. We end up with client libraries that just bail out if they encounter invalid JSON.

Reply:
What about quoted string?

Reply:
I would vote for placing such exceptional values into a differently named field, which is, however, impossible to do in this function. In any case, making a field sometimes (especially in rare cases) have a different data type (a string instead of a number) is bad, as it would cause crashes in type-aware clients that implicitly apply a schema.

In short, this function is not a proper place to make a policy on encoding non-numbers in places where numbers are expected.

Reply:
Is `null` any better?

Reply:
On paper, yes: with Rust and serde_json, one can directly serialize a "floating point or `null`" into `Option<f32>`.

In another possible opinion, any NaNs or infinities here should be treated as errors (i.e., `ceph_assert`) and all callers fixed up. Of course, this is a monumental task, that's why I am not suggesting this for real, but writing `null` and logging a backtrace might be something to consider.

Reply:
Very well, let's stick with `null`.